### PR TITLE
Refactor host names

### PR DESCRIPTION
**Pull request by eirslett**
The commit message explains the changes.

I only have access to a DigitalOcean account, so that's unfortunately the only place I have actually tested the setup...

What do you think about this refactoring?

### CODE REVIEW

File: terraform/openstack/hosts/main.tf [hunk 1]
```diff
@@ -8,10 +8,11 @@ variable net_id { }
 variable keypair_name { }
 variable image_name { }
 variable control_count {}
-variable resource_count {}
+variable worker_count {}
```
Comment (michaeltchapman) [inline terraform/openstack/hosts/main.tf]: Need to refactor L51 as well

    openstack_compute_instance_v2.worker: missing dependency: var.resource_count

**Comment by michaeltchapman** [comment]
I'm not sure the commit message matches the implementation. Do you mean to have short_name or datacenter in the name string?

I think `control-${count.index}.${var.datacenter}.${var.domain}` matches the commit message and might be nicer than short_name.

I like the idea of the fqdn for the hostname.

**Comment by eirslett** [comment]
Oops, sorry I missed the variable name! Should be fixed now.

My understanding of short_name is that it's the short name of a datacenter; e.g. "new-york-3" is the name of the datacenter, while "nyc3" is the short_name. (For now, it looks like 'mi' is the default short_name for everything, but we're using different short_names pr. datacenter, nyc0, nyc1 etc)

Is the intention of the short_name variable something else?
Of course, we can use var.datacenter in the host name instead of short_name, it's just a bit longer to type it. (But we might prevent some bugs coming from confusing the two)

**Comment by eirslett** [comment]
I'll close this one, maybe open other (more focused) PRs later.

**Pull request closed without merging**